### ISSUE TITLE
Basic access authentication

### ISSUE THREAD

**Issue by piotrkot** [yegor256/takes#505]
Could you please provide a sample of [basic authentication](https://en.wikipedia.org/wiki/Basic_access_authentication)?
What I am trying to do is an authenticating/secure resource. Initial access should result in server response with header "WWW-Authenticate: Basic realm...", that is prompt for credentials.
Provided correct credentials, it should be shown the resource.
It seems the below does the trick:
```
new FkRegex(
    "/secure",
    new TkAuth(
        new TkSecure(
           new TkText("secured")
        ),
        new PsBasic(
            "myrealm",
            new PsBasic.Default("mike password urn:users:michael")
        )
    )
)
```
but it does not.

**Comment by davvd** [comment]
@yegor256 dispatch this issue please, see [par.21](http://at.teamed.io/policy.html#21)

**Comment by piotrkot** [comment]
@davvd I see but please explain where I can get help with using the framework. I thought this is the way to ask for clarification.

**Comment by yegor256** [comment]
@piotrkot thanks for reporting. if something is not clear, it's definitely a bug. I will update our documentation now.

**Comment by piotrkot** [comment]
@yegor256 Thanks. Perhaps I could rephrase it to, say, Basic access authentication is not possible unless you prove with example otherwise. :)

**Comment by piotrkot** [comment]
@yegor256 In case the documentation is already updated, where can I find it?

**Comment by piotrkot** [comment]
@davvd @yegor256 Well, I has been a while. Do you have an update on this? I mean sorry for bugging you but you said you would dispatch it. Please, let me know if there is a problem with the fix.

**Comment by yegor256** [comment]
@piotrkot see 2124899 the test passes... it doesn't work for you?

**Comment by piotrkot** [comment]
@yegor256 Thanks for the test. This however does not work for me. The problem I see is in real-life scenario. Now, unauthenticated request for secured resource results in server returning Exception. But users will hardly ever wish to see the exception. Instead, they would expect to get a chance to authenticate. Thus, here I believe is preferable to get a server response with header "WWW-Authenticate: Basic realm...", that is a prompt for credentials.

**Comment by yegor256** [comment]
@piotrkot you're right, it was a bug. I reproduced it in 46c529dca11dc6ee3c0f67533ac26123b98780fe and fixed in 0488788d05cf69abf06d3c822d596986ae72a269

**Comment by yegor256** [comment]
@rultor release, tag is `0.32.5`

**Comment by piotrkot** [comment]
@yegor256 Yes, I also confirm the issue is fixed now. Thank you.

**Comment by yegor256** [comment]
@piotrkot please, close the ticket. thanks!
